flows/contact.go

```diff
@@ -46,20 +46,18 @@ func NewContact(
 	timezone *time.Location,
 	createdOn time.Time,
 	urns []urns.URN,
-	groups []assets.Group,
-	fields map[string]*Value) (*Contact, error) {
+	groups []*assets.GroupReference,
+	fields map[string]*Value,
+	missing assets.MissingCallback) (*Contact, error) {
 
-	urnList, err := ReadURNList(sa, urns, assets.IgnoreMissing)
+	urnList, err := ReadURNList(sa, urns, missing)
 	if err != nil {
 		return nil, err
 	}
 
-	groupList, err := NewGroupListFromAssets(sa, groups)
-	if err != nil {
-		return nil, err
-	}
+	groupList := NewGroupList(sa, groups, missing)
```

Comment:
I think we've run into problems because mailroom can create a contact on its side that has groups which, because they are inactive, aren't in the session assets... so for goflow to be sure that the contact only has fields and groups etc that exist in the session assets, it takes references and tries to resolve them, and potentially reports them as missing.